src/Compilers/CSharp/Portable/Compiler/MethodCompiler.cs

```diff
@@ -221,13 +221,42 @@ private static MethodSymbol GetEntryPoint(CSharpCompilation compilation, PEModul
                 !hasDeclarationErrors &&
                 !diagnostics.HasAnyErrors())
             {
-                var body = synthesizedEntryPoint.CreateBody();
+                BoundStatement body = synthesizedEntryPoint.CreateBody();
+
+                var dynamicAnalysisSpans = ImmutableArray<SourceSpan>.Empty;
+                var diagsForCurrentMethod = DiagnosticBag.GetInstance();
```

Comment:
Perhaps just pass `diagnostics` to `LowerBodyOrInitializer` instead, to ensure we don't drop diagnostics.